tools/perfgw.d

```diff
@@ -1,27 +1,16 @@
 /*
  * Trace all the guest submitted and completed IOs.
  */
-cdt*:::gw_read_start,
-cdt*:::gw_write_start,
-cdt*:::gw_flush_start
+cdt*:::gw-*-start
 {
     start[arg0] = timestamp;
 }
-cdt*:::gw_read_end
-/start[arg0] != 0/
+
+cdt*:::gw-*-done
+/start[arg0]/
 {
-    @time["read"] = quantize(timestamp - start[arg0]);
-    start[arg0] = 0;
-}
-cdt*:::gw_write_end
-/start[arg0] != 0/
-{
-    @time["write"] = quantize(timestamp - start[arg0]);
-    start[arg0] = 0;
-}
-cdt*:::gw_flush_end
-/start[arg0] != 0/
-{
-    @time["flush"] = quantize(timestamp - start[arg0]);
+    strtok(probename, "-");
+    this->cmd = strtok(NULL, "-");
+    @time[this->cmd] = quantize(timestamp - start[arg0]);
```

Comment:
I know I suggested this, but I'm not sure your previous way wasn't clearer. I leave it to you to decide what you prefer.